Sofa/framework/Type/src/sofa/type/Vec.h

```diff
@@ -569,10 +589,66 @@ class Vec : public sofa::type::fixed_array<ValueType,size_t(N)>
         return false;
     }
 
+
+    // operator[]
+    constexpr reference operator[](size_type i)
+    {
+        assert(i < N && "index in Vec must be smaller than size");
+        return elems[i];
+    }
+    constexpr const_reference operator[](size_type i) const
+    {
+        assert(i < N && "index in Vec must be smaller than size");
+        return elems[i];
+    }
+
+    // direct access to data
+    constexpr const ValueType* data() const noexcept
+    {
+        return elems.data();
+    }
+
+    constexpr typename ArrayType::iterator begin() noexcept
+    {
+        return elems.begin();
+    }
+    constexpr typename ArrayType::const_iterator begin() const noexcept
```

Comment:
```suggestion
    constexpr typename const_iterator begin() const noexcept
```
Can be applied to all `ArrayType::*`

Reply:
`typename` is also useless